src/cli/commands/global.js

```diff
@@ -90,11 +90,15 @@ async function getGlobalPrefix(config: Config, flags: Object): Promise<string> {
   } else {
     prefix = POSIX_GLOBAL_PREFIX;
   }
+
+  const binFolder = path.join(prefix, 'bin');
   try {
-    await fs.access(path.join(prefix, 'bin'), (nativeFs.constants || nativeFs).W_OK);
+    await fs.access(binFolder, (nativeFs.constants || nativeFs).W_OK);
   } catch (err) {
     if (err.code === 'EACCES') {
       prefix = FALLBACK_GLOBAL_PREFIX;
+    } else if (err.code === 'ENOENT') {
+      await fs.mkdirp(binFolder);
```

Comment:
should you just return prefix here since you call `mkdirp` again in the function below?